src/trace.jl

```diff
@@ -9,18 +50,17 @@ isprimitive(f) = f isa Core.Builtin || f isa Core.IntrinsicFunction
 
 const ignored_methods = [@which((1,2)[1])]
 
-@primitive ctx::Trace function (f::Any)(args...)
-  C, T = DynamicCall(f, args...), typeof.((f, args...))
-  (T ∈ ctx.seen || isprimitive(f) ||
-    method(C) ∈ ignored_methods ||
-    method(C).module ∈ [Core, Core.Inference]) && return f(args...)
-  push!(ctx.seen, T)
-  result = overdub(ctx, f, args...)
-  analyse((a...) -> ctx.warn(Warning(a...)), C)
-  return result
+function Cassette.prehook(ctx::TraceurCtx, f, args...)
```

Comment:
Previously this was (effectively) a `posthook`; best to keep it that way unless there's a specific reason to change it.

Reply:
What's the reasoning for that? I mean it's (probably) trivial to change, but why is `posthook` more appropriate here?

Reply:
The reasoning is that it tends to give warnings in the "right order", in the sense that type inference issues tend to propagate up the stack rather than down. e.g. in `f(x) = ...; g(x)`, `f` using a global variable won't affect the inferability of `g`, but `g` doing so will likely break `f` as well; so printing in post order gives you the root cause first followed by knock-on effects.

(The order to analysis passes is also designed to fit that rule, though it's obviously heuristic).

Reply:
Ah, makes sense.